autoload/clap/provider/bcommits.vim

```diff
@@ -5,13 +5,78 @@ let s:save_cpo = &cpoptions
 set cpoptions&vim
 
 let s:bcommits = {}
+let s:bcommits.syntax = 'clap_diff'
+
+let s:current = ''
+let s:begin = '^[^0-9]*[0-9]\{4}-[0-9]\{2}-[0-9]\{2}\s\+'
+let s:shas = []
 
 function! s:bcommits.source() abort
-  call g:clap.abort('Not implemented yet')
+  let s:shas = split(system('git log --format=format:%h'), "\n")
+  let s:git_root = clap#path#get_git_root()
+  if empty(s:git_root)
+    call g:clap.abort('Not in git repository')
+    return
+  endif
+
+  let s:current = bufname(g:clap.start.bufnr)
+  if empty(s:current)
+    return ['The current buffer is not in the working tree' . s:current]
+  else
+    call system('git show '.s:current.' 2> '.(has('win32') ? 'nul' : '/dev/null'))
+  endif
+  let s:source = "git log '--color=never' '--date=short' '--format=%cd %h%d %s (%an)' '--follow' '--' ".s:current
+  return split(system(s:source), "\n")
```

Comment:
Actually you can directly return the system command, so that some native optimization of vim-clap can be used.

```suggestion
  return s:source
```

Same to the `source` of `commits` provider, the old code is not optimal.

```diff
+  return source
-  return split(system(source), "\n")
```

Reply:
Not sure if I missed anything here. After changed to 
```vim
let s:source = "git log '--color=never' '--date=short' '--format=%cd %h%d %s (%an)' '--follow' '--' ".s:current
return s:source
```
got an error message

```
Error detected while processing function <SNR>268_on_event[11]..<SNR>268_on_complete_maple:                         
line   22:                                                                                                          
E716: Key not present in Dictionary: lines, decoded.total)  
```
Are there any dependency?

Reply:
Are you using the latest vim-clap? Can't reproduce. Should work even without the Rust binary, so I would say there is no dependency here.

Reply:
Yes. I remove the folder and pull the vim-clap again. It works fine now.